ksqldb-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlAvroSerializerTest.java

```diff
@@ -188,35 +231,46 @@ public class KsqlAvroSerializerTest {
           + "\"fields\":["
           + "{\"name\":\"key\",\"type\":[\"null\",\"string\"],\"default\":null},"
           + "{\"name\":\"value\",\"type\":[\"null\",\"int\"],\"default\":null}],"
-          + "\"connect.internal.type\":\"MapEntry\"}}");
+          + "\"connect.internal.type\":\"MapEntry\"},"
+          + "\"connect.name\":\"io.confluent.ksql.avro_schemas.KsqlDataSourceSchema\"}"
+          + "}");
 
   private static final org.apache.avro.Schema DECIMAL_SCHEMA =
       parseAvroSchema(
           "{"
               + "\"type\": \"bytes\","
               + "\"logicalType\": \"decimal\","
               + "\"precision\": 4,"
-              + "\"scale\": 2"
+              + "\"scale\": 2,"
+              + "\"connect.version\": 1,"
+              + "\"connect.parameters\":{\"scale\":\"2\",\"connect.decimal.precision\":\"4\"},"
+              + "\"connect.name\":\"org.apache.kafka.connect.data.Decimal\""
```

Comment:
What would be the **REAL purpose** of this test? 
The latest version for the old existing tables/streams has NO `connect metadata`. This fix does not add any more schema (with or without `connect metadata`) to the SR. Therefore, if data is serialised by an old schema without `connect metadata`, that will be always deserialised with the same schema.
rmr: Based on our agreement, I added `connect metadata` to the schemas of these tests in order to fix the broken tests. What you suggest is not aligned with our former discussions.

Reply:
The purpose is to test upgrade path of new deserializer with `connect metadata` can deserialize old data serialized without `connect metadata`. I think we agreed you will add flag to serde factory class to control whether use `connect metadata` or not and test this upgrade path. Maybe the test belong to integration test. I even see you have a commit to add the flag: https://github.com/confluentinc/ksql/pull/9238/commits/cdd78607f858e69cf0f28ee8d19e95a36686d356. But it's gone now.

Reply:
`serde factory class` is Immutable. Adding any flag to it breaks Immutability tests.

Reply:
You need to make `private boolean connectMetadata = true;` to be `private final boolean connectMetadata;` to not break the immutability tests

Reply:
final variables van not be set to other values then!

Reply:
About deserialisation tests: The tests already serialise data objects using schemas with no `connect metadata `and deserialise them using schemas with `connect metadata`.  For example in `KsqlAvroDeserializerTest.shouldDeserializeAvroDecimal()`, the `deserializer` object  has `connect metadata` as it is created using `KsqlAvroSerdeFactory` while the `serialaizing` through `givenConnectSerialized()` method creates serializer with no `connect metadata`. Therefore, luckily the test is already what we want! It is the same with other data types.
P.S. By the way, if a variable is defined as `final` in the Immutable `KsqlAvroSerdeFactory` class, how we can set it to `false` (in order to set the `CONNECT_META_DATA_CONFIG` to `false`)  later?

Reply:
Looks like https://github.com/confluentinc/ksql/blob/5940036e449bd48d25a530956fae0d48cf905c05/ksqldb-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlAvroDeserializerTest.java#L224-L225 is used in `KsqlAvroDeserializerTest` unit test to serialize data. I don't see where `CONNECT_META_DATA_DEFAULT` is configured to false and the default is `true`. Am I missing anything?

> P.S. By the way, if a variable is defined as final in the Immutable KsqlAvroSerdeFactory class, how we can set it to false (in order to set the CONNECT_META_DATA_CONFIG to false) later?

Can you create two `KsqlAvroSerdeFactory` object with different flags?

Reply:
I did create  two KsqlAvroSerdeFactory objects with different flags as you can see in this [commit](https://github.com/confluentinc/ksql/commit/cdd78607f858e69cf0f28ee8d19e95a36686d356), but then `connectMetadata` can not be defined as final.

Reply:
> Looks like
> 
> https://github.com/confluentinc/ksql/blob/5940036e449bd48d25a530956fae0d48cf905c05/ksqldb-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlAvroDeserializerTest.java#L224-L225
> 

Maybe a better solution is to turn off `connect metadata` off here in the `configs`. That would less expensive than modifying `KsqlAvroSerdeFactory` class.

Reply:
> The purpose is to test upgrade path of new deserializer with `connect metadata` can deserialize old data serialized without `connect metadata`. 

By the way, I am against adding these tests since old serialised data (with NO `connect metadata`), will be deserialised with the latest schema in SR (the one with NO `connect metadata`). This fix does not change anything about the old existing tables and streams